content/articles/javascript-math-anim/index.md

```diff
@@ -0,0 +1,207 @@
+Animations are a great way to catch a user's attention and enhance the look of your website. In this article, we will look at how Javascript's built-in functions such as `Math.cos()` and `Math.sin()` can be applied to create the animations. We will be creating an animation to show a simple rotating system so that we see how the functions are applied. The final output will resemble this:
+
+![Circle](/engineering-education/javascript-math-anim/circle.png)
+
+Where the line shows the path the image follows.
+
+### Prerequisites
+
+1. An understanding of JavaScript
+2. An understanding of HTML and CSS
+
+### A brief look at the sin and cos functions in creating a circle
+
+These two functions play a great role in defining the x and y coordinates of a circle. Let's have a look at this figure.
+
+![Triangle shot](/engineering-education/javascript-math-anim/triangle.png)
+
+Using the basic trigonometric formulas, we see that `sin θ = (y/r)` and `cos θ = (x/r)`.
+
+Based on these, it's clear that given a circle of radius `r` at an angle of θ, the coordinates of the point (x, y)
+are found using:
+
+* x = r cos(θ)
+* y = r sin(θ)
+
+We will use these two functions to rotate an image in our animation. Find more about these [here](http://www.opentextbookstore.com/trig/trig-5-3.pdf).
+
+### The JavaScript code
+
+This is the JavaScript code.
+
+```javascript
+// The image variable
+let image = document.querySelector("img");
+// 90 degrees
+let angle = Math.PI / 2;
+
+function animate(currentTimeParam, prevTimeArg) {
+
+    if (prevTimeArg != null) {
+        angle += (currentTimeParam - prevTimeArg) * 0.004;
+    }
+    // Setting the previous time to the time the function currently fires
+    prevTimeArg = currentTimeParam ;
+    image.style.top = (Math.sin(angle) * 150) + "px";
+    image.style.left = (Math.cos(angle) * 150) + "px";
+    // Passing the current time and previous times
+    // For the previous time, we pass in the `prevTimeArg` argument
+    // prevTimeArg` stores this time
```

Comment:
A bit of a misunderstanding here maybe I wasn't clear enough. I meant that I wanted you to add a comment regarding what arguments are passed to `animate` in the first call to `requestAnimationFrame` at line 52: 

```javascript 
requestAnimationFrame(animate)
```
At this line, `animate` is passed as a callback function. If you look at the declaration for ```requestAnimationFrame``` it expects a callback function with one argument for the time but your animate function takes two arguments. I can assume that requestAnimationFrame simply gives `null` as the second argument to `animate` here but I think this can be confusing to readers. 

As for the comments you recently added at lines 48-49, I thought it was already clear what was going on there and you may remove that.

Please let me know if you don't understand what I'm saying here.